src/services/userService.js

```diff
@@ -0,0 +1,45 @@
+let users = [];
+
+const resetUsers = () => {
+  users = [];
+};
+
+const getAllUsers = () => {
+  return users;
+};
+
+const getUserById = (id) => {
+  return users.find((person) => person.id === +id) || null;
+};
+
+const createUser = (name) => {
+  const user = {
+    id: Math.floor(Math.random() * 1000),
```

Comment:
Bug: The way you are generating ids for the users can lead to collisions. Two different users might get the same id if the random function returns the same number. This can lead to unexpected behavior in your application. It would be better to use a safe way to generate unique ids, like a UUID library or maintaining a counter that you increment each time.